components/landing/topCollections/utils/useTopCollections.ts

```diff
@@ -24,30 +24,40 @@ const proccessData = (
   collectionsList: CollectionEntity[],
   collectionsSales: CollectionSales[],
 ) => {
-  return collectionsList.map((e): CollectionEntityWithVolumes => {
-    const thisCollectionSales = collectionsSales.find(
-      ({ id }) => id === e.id,
-    ) as CollectionSales
-    const saleEvents = thisCollectionSales.sales.map((nft) => nft.events).flat()
+  return Promise.all(
+    collectionsList.map(async (e): Promise<CollectionEntityWithVolumes> => {
+      const thisCollectionSales = collectionsSales.find(
+        ({ id }) => id === e.id,
+      ) as CollectionSales
+      const saleEvents = thisCollectionSales.sales
+        .map((nft) => nft.events)
+        .flat()
 
-    return {
-      ...e,
-      image: sanitizeIpfsUrl(e.image),
-      averagePrice: calculateAvgPrice(e.volume as string, e.buys),
-      volume: volume(saleEvents),
-      weeklyVolume: weeklyVolume(saleEvents),
-      monthlyVolume: monthlyVolume(saleEvents),
-      threeMonthVolume: threeMonthlyVolume(saleEvents),
-      weeklyrangeVolume: weeklyrangeVolume(saleEvents),
-      monthlyrangeVolume: monthlyrangeVolume(saleEvents),
-      threeMonthlyrangeVolume: threeMonthRangeVolume(saleEvents),
-    }
-  })
+      const image = e.image
+        ? sanitizeIpfsUrl(e.image)
+        : sanitizeIpfsUrl(
+            getCollectionImage(await processSingleMetadata(e.metadata)) || '',
+          )
```

Comment:
why?

Reply:
image is null

Reply:
If it is because non-koda collections i will remove them.

Reply:
as you can see `image` is null , `TopCollectionsCard` gets the images from the metadata

query 

`````
query topCollections($limit: Int, $orderBy: CollectionEntityOrderByInput = volume_DESC, $where: CollectionEntityWhereInput = {volume_gt: "0"}) {
  collectionEntities(orderBy: [$orderBy], limit: $limit, where: $where) {
    id
    name
    image
    media
    volume
    metadata
    floor
    nftCount
    ownerCount: distribution
    __typename
  }
}


{
  "orderBy": "volume_DESC",
  "where": {
    "issuer_not_in": []
  },
  "limit": 12
}
`````
results

`````
{
  "data": {
    "collectionEntities": [
      {
        "id": "0x3c93690bbe585475fdfadab3f59b4604008c7ac4",
        "name": "Bloquinhos",
        "image": null,
        "media": null,
        "volume": "0",
        "metadata": "https://arweave.net/brcg0wOzdEIjwBswDdt2nTjsl0-8FJNtQaiKCwaOMqQ ",
        "floor": "0",
        "nftCount": 0,
        "ownerCount": 0,
        "__typename": "CollectionEntity"
      },
      {
        "id": "0x25194dfc7981d8a13367fe19b5b1c5fc010d535f",
        "name": "Conjunto",
        "image": null,
        "media": null,
        "volume": "0",
        "metadata": "https://arweave.net/btZcKjV2qZ5OGxQm02h6rCcK1MY6bV6JYKV30Vr1ipM",
        "floor": "0",
        "nftCount": 0,
        "ownerCount": 0,
        "__typename": "CollectionEntity"
      },
      {
        "id": "0x3c549c45588fb42a94ca7b84081df43d95952c01",
        "name": "Memories",
        "image": null,
        "media": null,
        "volume": "0",
        "metadata": "https://arweave.net/UIV7b488550HzoAdpwRO7laTUDq9VPOD0YEP8sQfkaU",
        "floor": "0",
        "nftCount": 0,
        "ownerCount": 0,
        "__typename": "CollectionEntity"
      },
      {
        "id": "0x1b60a7ee6bba284a6aafa1eca0a1f7ea42099373",
        "name": "Vortices",
        "image": null,
        "media": null,
        "volume": "0",
        "metadata": "ipfs://bafkreicthpidyo3gznp3uuezvweiur7xcbo5qcwgw3x4teui2psoqxxbgq",
        "floor": "0",
        "nftCount": 0,
        "ownerCount": 0,
        "__typename": "CollectionEntity"
      },
      {
        "id": "0x0b6504d95f9d550d274468fb6de5d13b7e64aa6a",
        "name": "Higher",
        "image": null,
        "media": null,
        "volume": "0",
        "metadata": "ipfs://bafkreickjpqtqulirvugyqupeogbvum2uqsrp5focif5427fd2fzgywo4u",
        "floor": "0",
        "nftCount": 0,
        "ownerCount": 0,
        "__typename": "CollectionEntity"
      },
      {
        "id": "0xc029b380f8a451cfd9e5124fa9fcad4397b8c119",
        "name": "Unfoldings",
        "image": null,
        "media": null,
        "volume": "0",
        "metadata": "ipfs://bafkreib5pt53graj4wexhpgj3qrl37m3bsgjf3hz5muogqccdtutaq2huq",
        "floor": "0",
        "nftCount": 0,
        "ownerCount": 0,
        "__typename": "CollectionEntity"
      }
    ]
  }
}

`````